rfcs/0180-broken-package-removal.md

```diff
@@ -0,0 +1,74 @@
+---
+feature: broken-package-removal
+start-date: 2024-07-14
+author: Jörg "Mic92" Thalheim
+co-authors:
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+This RFCs defines under what conditions we remove broken and unmaintained leaf packages.
+The RFC does not have the ambition to define all reasons to remove a package but
+instead focuses on simple rules that can be automated.
+
+# Motivation
+[motivation]: #motivation
+
+Broken and unmaintained packages still cost us valuable time.
+Broken packages still have to be evaluated to some extend
+and unmaintained packages might still have to rebuild when one of their dependencies changes.
+Because it is easy to add packages, it should be also easy to remove them, ideally automated
+
+# Detailed design
+[design]: #detailed-design
+
+## Broken Packages
+
+Packages that are marked as broken on all platforms should be removed after one full NixOS release cycle.
```

Comment:
```suggestion
Packages that are marked as broken unconditionally should be removed after one full NixOS release cycle.
```

Example: a package that does not build with the most recent Python is technically working on all platforms where the older Python is present.

Reply:
Sound good. Thanks.